spock-tapestry/src/main/java/org/spockframework/tapestry/TapestryExtension.java

```diff
@@ -85,29 +97,58 @@
  * @author Peter Niederwieser
  */
 public class TapestryExtension extends AbstractGlobalExtension {
-  public void visitSpec(SpecInfo spec) {
+
+  // since Tapestry 5.4
+  @SuppressWarnings("unchecked")
+  private static final Class<? extends Annotation> importModuleAnnotation =
+      (Class) ReflectionUtil.loadClassIfAvailable("org.apache.tapestry5.ioc.annotations.ImportModule");
+
+  // deprecated as of Tapestry 5.4
+  @SuppressWarnings("unchecked")
+  private static final Class<? extends Annotation> submoduleAnnotation =
+      (Class) ReflectionUtil.loadClassIfAvailable("org.apache.tapestry5.ioc.annotations.SubModule");
+
+  @Override
+  public void visitSpec(final SpecInfo spec) {
     Set<Class<?>> modules = collectModules(spec);
-    if (modules == null) return;
+    if (modules == null) {
+      return;
+    }
 
     IMethodInterceptor interceptor = new TapestryInterceptor(spec, modules);
     spec.addSharedInitializerInterceptor(interceptor);
     spec.addInitializerInterceptor(interceptor);
     spec.addCleanupSpecInterceptor(interceptor);
   }
 
-  // Returns null if no SubModule annotation was found.
-  // Returns an empty list if one or more SubModule annotations were found,
+  // Returns null if no ImportModule or SubModule annotation was found.
+  // Returns an empty list if one or more ImportModule or SubModule annotations
+  // were found,
   // but they didn't specify any modules. This distinction is important to
   // allow activation of the extension w/o specifying any modules.
-  private Set<Class<?>> collectModules(SpecInfo spec) {
+  private Set<Class<?>> collectModules(final SpecInfo spec) {
     Set<Class<?>> modules = null;
-
     for (SpecInfo curr : spec.getSpecsTopToBottom()) {
-      SubModule subModule = curr.getAnnotation(SubModule.class);
-      if (subModule == null) continue;
-
-      if (modules == null) modules = new HashSet<Class<?>>();
-      modules.addAll(Arrays.<Class<?>>asList(subModule.value()));
+      if (importModuleAnnotation != null && spec.isAnnotationPresent(importModuleAnnotation)){
+        org.apache.tapestry5.ioc.annotations.ImportModule importModule = curr
+            .getAnnotation(org.apache.tapestry5.ioc.annotations.ImportModule.class);
```

Comment:
You can't use the class-literal here, you need to use the `importModuleAnnotation` otherwise you create a hard dependency on it and it will fail with NoClassDefFoundError.
You'll also need to use reflection to invoke the `getValue()` method on the annotation instance.